openmls/src/group/core_group/create_commit.rs

```diff
@@ -60,20 +92,19 @@ impl CoreGroup {
         // TODO: #581 Filter proposals by support
         // 11.2:
         // Proposals with a non-default proposal type MUST NOT be included in a commit
-        // unless the proposal type is supported by all the members of the group that will
-        // process the Commit (i.e., not including any members being added or removed by
-        // the Commit).
+        // unless the proposal type is supported by all the members of the group that
+        // will process the Commit (i.e., not including any members being added
+        // or removed by the Commit).
 
         let proposal_reference_list = proposal_queue.commit_list();
 
         // Make a copy of the current tree to apply proposals safely
         let mut diff: TreeSyncDiff = self.treesync().empty_diff()?;
 
-        // If this is not an external commit we have to set our own leaf index
-        // and add our leaf. Also, we have to generate the
-        // [`KeyPackageBundlePayload`] slightly differently, because we can't
-        // just pull it from the tree if we're a `NewMember`.
-        let key_package_bundle_payload = self.prepare_kpb_payload(backend, &params, &mut diff)?;
+        // If this is not an external commit we have to set our own leaf index manually
+        if params.commit_type() == CommitType::External {
```

Comment:
The comment contradicts the `if` statement here.

Reply:
Good point. The negation shouldn't be there.